files/solr_check_instance.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+# Step through all cores under one instance and throw an exit code if something is not running.
+# Exit code 3: possibly not loaded yet, try waiting a bit
+# Exit code 4: hard fail, no need to wait any more
+
+set -euo pipefail
+
+if [ "$#" -lt "1" ]; then
+	echo "Cowardly refusing to reload all cores on this server. Please specify an instance as the first param."
+	exit 2
+fi
+
+instance="${1}v4"
+target="*/${instance}/*"
+
+status=$(env -i curl "http://localhost:8080/$instance/admin/cores?wt=json" -sS)
+for core in $(find /var/lib/solr -maxdepth 2 -mindepth 2 -path "$target" -type d -not -name logs); do
+	if [[ ! -f "${core}/core.properties" ]]; then continue; fi
+
+	index=$(basename $core)
+	echo "Checking '$instance/$index'"
+	failure=$(echo "$status" | jq ".initFailures[\"$index\"]")
```

Comment:
Not sure if these scripts are used on CWP or not but CWP doesn't have `jq` installed

Reply:
Thanks, I've added a require for the `jq` package.
The script was added in the `puppet-upgrade`, but I've created the buster branch of puppet-ss_solr from `master` to see all changes.